staging/src/k8s.io/legacy-cloud-providers/azure/azure_client.go

```diff
@@ -973,6 +976,36 @@ func (az *azVirtualMachineScaleSetsClient) List(ctx context.Context, resourceGro
 	return result, nil
 }
 
+func (az *azVirtualMachineScaleSetsClient) CreateOrUpdate(ctx context.Context, resourceGroupName string, vmScaleSetName string, parameters compute.VirtualMachineScaleSet) (resp *http.Response, err error) {
+	/* Write rate limiting */
+	if !az.rateLimiterWriter.TryAccept() {
+		err = createRateLimitErr(true, "NiCreateOrUpdate")
+		return
+	}
+
+	klog.V(10).Infof("azVirtualMachineScaleSetsClient.CreateOrUpdate(%q,%q): start", resourceGroupName, vmScaleSetName)
+
+	klog.V(3).Infof("azVirtualMachineScaleSetsClient.CreateOrUpdate(%q,%q): verify the status of the vmss being created or updated", resourceGroupName, vmScaleSetName)
+	vmss, err := az.Get(ctx, resourceGroupName, vmScaleSetName)
+	if vmss.ProvisioningState != nil && strings.EqualFold(*vmss.ProvisioningState, virtualMachineScaleSetsDeallocating) {
```

Comment:
This would introduce a new API call. Let's move the check out of vmssClient and do the check before invoking `CreateOrUpdate()`.